pages/api/submit.tsx

```diff
@@ -191,4 +245,105 @@ const processFormData = async (
   }
 };
 
+const submitToListManagementAPI = async (
+  irccConfig: IRCCConfig,
+  listManagerHost: string,
+  listManagerApiKey: string,
+  reqFields: Responses,
+  form: PublicFormSchemaProperties,
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  // get program and language values from submission using the ircc config refrenced ids
+  const programs = JSON.parse(reqFields[irccConfig.programFieldID] as string) as CheckboxValue;
+  const programList = programs.value;
+
+  const languages = JSON.parse(reqFields[irccConfig.languageFieldID] as string) as CheckboxValue;
+
+  const languageList = languages.value;
+
+  const contact = reqFields[irccConfig.contactFieldID] as string;
+
+  // get the type of contact field from the form template
+  const contactFieldFormElement = form.elements.filter(
+    (value) => value.id === irccConfig.contactFieldID
+  );
+
+  const contactFieldType =
+    contactFieldFormElement.length > 0
+      ? contactFieldFormElement[0].properties.validation?.type
+      : false;
+
+  if (contactFieldType) {
+    // forEach slower than a for loop https://stackoverflow.com/questions/43821759/why-array-foreach-is-slower-than-for-loop-in-javascript
+    // yes its a double loop O(n^2) but we know n <= 4
+    for (let i = 0; i < languageList.length; i++) {
+      for (let j = 0; j < programList.length; j++) {
+        const language = languageList[i];
+        const program = programList[j];
+        let listID;
+        // try getting the list id from the config json. If it doesn't exist fail gracefully, log the message and send 500 error
+        // 500 error because this is a misconfiguration on our end its not the users fault i.e. 4xx
+        try {
+          listID = irccConfig.listMapping[language][program][contactFieldType];
+        } catch (e) {
+          logMessage.error(
+            `IRCC config does not contain the following path ${language}.${program}.${contactFieldType}`
+          );
+          return res.status(500).json({
+            error: `IRCC config does not contain the following path ${language}.${program}.${contactFieldType}`,
+          });
+        }
+
+        let response: AxiosResponse<never>;
+        try {
+          // Now we create the subscription
+          response = await axios.post(
+            `${listManagerHost}/subscription`,
+            {
+              [contactFieldType]: contact,
+              list_id: listID,
+            },
+            {
+              headers: {
+                Authorization: listManagerApiKey,
+              },
+            }
+          );
+        } catch (e) {
+          logMessage.error(
+            `Subscription failed with status ${e.response.status} and message ${e.response.data}`
+          );
+          logMessage.error(e);
+          return res.status(500).json({
```

Comment:
As long as we're logging the error internally we can just return the same message to let the client know it failed along the way.
`return res.status(500).json({ received: false });`